### PR TITLE
[SPARK-14072][core] Show JVM/OS version information when we run a benchmark program

### PR DESCRIPTION
**Pull request by kiszk**
## What changes were proposed in this pull request?

This PR allows us to identify what JVM is used when someone ran a benchmark program. In some cases, a JVM version may affect performance result. Thus, it would be good to show processor information and JVM version information.

```
model name  : Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
JVM information : OpenJDK 64-Bit Server VM, 1.7.0_65-mockbuild_2014_07_14_06_19-b00
Int and String Scan:                Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
-------------------------------------------------------------------------------------------
SQL Parquet Vectorized                    981 /  994         10.7          93.5       1.0X
SQL Parquet MR                           2518 / 2542          4.2         240.1       0.4X
```

```
model name  : Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
JVM information : IBM J9 VM, pxa6480sr2-20151023_01 (SR2)
String Dictionary:                  Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
-------------------------------------------------------------------------------------------
SQL Parquet Vectorized                    693 /  740         15.1          66.1       1.0X
SQL Parquet MR                           2501 / 2562          4.2         238.5       0.3X
```
## How was this patch tested?

Tested by using existing benchmark programs

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by kiszk** [comment]
Jenkins, retest this please

**Comment by rxin** [comment]
Maybe change it to

```
CPU: ...
JVM: ...
```

Can we add OS version too? And at some point maybe we don't need to have 3 lines, but rather one or two lines.

**Comment by kiszk** [comment]
I updated like this

```
CPU   : Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
JVM,OS: OpenJDK 64-Bit Server VM 1.7.0_65-mockbuild_2014_07_14_06_19-b00, Linux 2.6.32-504.el6.x86_64
SQL Single Int Column Scan:         Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
-------------------------------------------------------------------------------------------
SQL Parquet Vectorized                    568 /  598         27.7          36.1       1.0X
SQL Parquet MR                           1784 / 1794          8.8         113.4       0.3X
```

**Comment by kiszk** [comment]
Jenkins, retest this please

**Comment by rxin** [comment]
This is a bit ugly. Actually since it is obvious what the cpu/jvm/os are, why don't we just do

```
OpenJDK 64-Bit Server VM 1.7.0_65-mockbuild_2014_07_14_06_19-b00 on Linux 2.6.32-504.el6.x86_64
Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
```

File: core/src/main/scala/org/apache/spark/util/Benchmark.scala [hunk 1]
```diff
@@ -91,16 +92,31 @@ private[spark] object Benchmark {
    * This should return something like "Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz"
```
Comment (holdenk) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: nit: Maybe update the comment to to reflect the new format (e.g. "CPU   :Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz")

Comment (rxin) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: nah i don't think we need to do those. These are meant as a snapshot for reference.

Comment (rxin) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: We also don't know what JVM or OS version they were run on.

**Comment by kiszk** [comment]
I see. Here is the current format as you suggested.

```
OpenJDK 64-Bit Server VM 1.7.0_65-mockbuild_2014_07_14_06_19-b00 on Linux 2.6.32-504.el6.x86_64
Intel(R) Xeon(R) CPU E5-2697 v2 @ 2.70GHz
SQL Single Int Column Scan:         Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
-------------------------------------------------------------------------------------------
SQL Parquet Vectorized                    604 /  687         26.0          38.4       1.0X
SQL Parquet MR                           1775 / 1788          8.9         112.9       0.3X
```

File: core/src/main/scala/org/apache/spark/util/Benchmark.scala [hunk 2]
```diff
@@ -91,16 +92,31 @@ private[spark] object Benchmark {
...
     } else if (SystemUtils.IS_OS_LINUX) {
       Try {
         val grepPath = Utils.executeAndGetOutput(Seq("which", "grep"))
         Utils.executeAndGetOutput(Seq(grepPath, "-m", "1", "model name", "/proc/cpuinfo"))
+        .replaceFirst("model name[\\s*]:[\\s*]", "")
       }.getOrElse("Unknown processor")
     } else {
       System.getenv("PROCESSOR_IDENTIFIER")
     }
+    s"${cpu}"
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/util/Benchmark.scala]: this is just `cpu` instead of `s"${cpu}"`

**Comment by rxin** [comment]
LGTM - please fix the minor comment and scala style.

**Comment by rxin** [comment]
Thanks - merging in master.

**Pull request closed without merging**